### DISCUSSION TITLE
Drawing a "multi" geometry shape in osgEarth

### DISCUSSION THREAD

**Discussion by Warren-Whitfield** [pelicanmapping/osgearth#2486] [Support Q&A] [answered]
Hi All
I am relatively new to osgEarth. I want to draw an outline (or filled) polygon type shape consisting of lines and arcs. What is the best way to do this please?

**Comment by gwaldron** [comment]
You are looking for Annotations.

Look at the example earth file `annotation.earth` in the `tests` folder. The "Draped Polygon" annotation should do what you want.

If you need to do it in code, take a look at the `osgearth_annotation.cpp` example app; lots of examples in there.

**Comment by Warren-Whitfield** [comment]
Cool thanks Glenn that works ok. (I needed to do it in code by the way)

But, I don't have these functions as used in the osgearth_annotation.cpp example line 366 onwards.
 _// highlight annotation upon hover by default:
 DecorationInstaller highlightInstaller("hover", new HighlightDecoration());_

I'm using vcpkg and did a complete search of file contents in the folder and I definitely don't have these.

Any thoughts?

Additionally, so currently I have an outline of two lines and two arcs representing a range/azimuth selection sector. The primitives are in essence all seperate entities. So how would I fill this area? In reality, I guess I want this to be some sort of polygon or enclosed area. (![Screenshot from 2024-04-03 11-31-40](https://github.com/gwaldron/osgearth/assets/159110675/5b0bb472-1c78-4026-a981-f5d336f56cef)

**Reply by gwaldron** [reply]
Exactly, you are going to have to make a polygonal feature if you want to fill it.

There is no such class DecorationInstaller, are you using an old version of osgEarth?

**Comment by Warren-Whitfield** [comment]
Ok, I'll try that, thanks

That extract was from the  osgearth_annotation.cpp example?

**Reply by gwaldron** [reply]
What version of osgEarth are you using? 3.5 is the latest.

**Comment by Warren-Whitfield** [comment]
The version from vcpkg is as follows:-
osgearth:x64-linux  3.4#1

**Comment by gwaldron** [comment]
I had to go all the way back to Version 2.7 to find a reference to `DecorationInstaller`. 

https://github.com/gwaldron/osgearth/blob/osgearth-2.7/src/applications/osgearth_annotation/osgearth_annotation.cpp#L492

That is 9 years ago :)

**Comment by Warren-Whitfield** [comment]
Sh!t, my apologies, its just what came up when I first searched osgEarth examples and I've never bothered to check the version. Sorry about that!

So re my previous query, how do I get that sector into a polygon to be able to fill it? It consists of 2 LineStrings and 2 CircleNodes.

**Reply by gwaldron** [reply]
:)
There's no function in osgEarth to union geometries or nodes like that, so you're going to have to make a FeatureNode or LocalGeometryNode with your custom geometry. Good luck!

**Comment by Warren-Whitfield** [comment, marked as answer]
So, I've decided to just create my own arc's using a series of line segments. That seemed the simplest solution to me to get everything into a polygon so that it can be filled if required.
